twoliter/src/lock.rs

```diff
@@ -171,8 +330,8 @@ impl Lock {
                     (image.name.clone(), image.vendor.clone()),
                     image.version.clone(),
                 );
-                let locked_image = LockedImage::new(vendor, image);
-                let kit = Self::find_kit(vendor, &locked_image).await?;
+                let mut locked_image = LockedImage::new(vendor, image, "");
+                let kit = Self::find_kit(vendor, &mut locked_image).await?;
```

Comment:
nit: mutating the object after creation is a strange contract that could lead to bugs in the future -- the state of a `LockedImage` can't be known without context. I'm guessing we do this so that we can take advantage of `LockedImage` providing us the source URI?

Can we do something here to make the OO contract stronger?